domehunter/dome_control.py

```diff
@@ -0,0 +1,674 @@
+        # update the debug LEDs
+        if self.last_direction == "CCW":
+            self._turn_led_off(leds=['relay_2_normally_closed'])
+        self._turn_led_on(leds=['relay_2_normally_open'])
+        # turn on rotation
+        self.rotation_relay.on()
+        self._dome_moving = True
+        # update the rotation relay debug LEDs
+        self._turn_led_on(leds=['relay_1_normally_open'])
+        self._turn_led_off(leds=['relay_1_normally_closed'])
+        cmd_status = True
+        return cmd_status
+
+    def _move_ccw(self):
+        """
+        Set dome to move counter-clockwise.
+
+        Returns
+        -------
+        integer
+            Command status return code (tbd).
+
+        """
+        # if testing, deactivate the home_sernsor_pin to simulate leaving home
+        if self.testing and self._at_home:
+            self.home_sensor_pin.drive_low()
+        # update the last_direction instance variable
+        self.last_direction = self.current_direction
+        # now update the current_direction variable to CCW
+        self.current_direction = "CCW"
+        # set the direction relay switch to CCW position
+        self.direction_CW_relay.off()
+        self._dome_moving = False
+        # update the debug LEDs
+        if self.last_direction == "CW":
+            self._turn_led_off(leds=['relay_2_normally_open'])
+        self._turn_led_on(leds=['relay_2_normally_closed'])
+        # turn on rotation
+        self.rotation_relay.on()
+        # update the rotation relay debug LEDs
+        self._turn_led_on(leds=['relay_1_normally_open'])
+        self._turn_led_off(leds=['relay_1_normally_closed'])
+        cmd_status = True
+        return cmd_status
+
+    def _stop_moving(self):
+        """
+        Stop dome movement by switching the dome rotation relay off.
+        """
+        self.rotation_relay.off()
+        self._dome_moving = False
+        # update the debug LEDs
+        self._turn_led_off(leds=['relay_1_normally_open'])
+        self._turn_led_on(leds=['relay_1_normally_closed'])
+        # update last_direction with current_direction at time of method call
+        self.last_direction = self.current_direction
+
+    def _simulate_ticks(self, num_ticks):
+        """
+        Method to simulate encoder ticks while in testing mode.
+        """
+        tick_count = 0
+        # repeat this loop of driving the mock pins low then high to simulate
+        # an encoder tick. Continue until desired number of ticks is reached.
+        while tick_count < num_ticks:
+            self.encoder_pin.drive_low()
+            # test_mode_delay_duration is set so that it will always exceed
+            # the set bounce_time
+            # of the pins
+            time.sleep(self.test_mode_delay_duration)
+            self.encoder_pin.drive_high()
+            time.sleep(self.test_mode_delay_duration)
+            tick_count += 1
+
+    def _turn_led_on(self, leds=[]):  # pragma: no cover
```

Comment:
It seems like there is a lot could that potentially raise an exception in these led methods. In the above movement methods you are turning these on and off a bunch during movement (e.g. during the incrementing of the encoder) and there is not a ton of exception handling. Seems like a dangerous situation where the leds cause exceptions and the dome either doesn't rotate or keeps rotating forever.